src/granite_io/backend/openai.py

```diff
@@ -46,8 +46,13 @@ def __init__(self, config: aconfig.Config):
 
     async def generate(self, inputs: GenerateInputs):
         """Run a direct /completions call"""
+
+        # Implementations of the OpenAI APIs tend to get upset if you pass them null
+        # values, particularly if those values are associated with arguments that the
+        # implementation doesn't support. So strip out null parameters.
+        args_dict = {k: v for k, v in inputs.model_dump().items() if v is not None}
```

Comment:
```suggestion
        args_dict = inputs.model_dump(exclude_none=True)
```
Or if you want to allow someone choosing to set None, then use exclude_defaults=True instead (leaves out defaults, of course).

Reply:
Code that sets `exclude_defaults=True` would break if we added sensible defaults for any parameters.